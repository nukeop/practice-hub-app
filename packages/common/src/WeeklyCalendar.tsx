'use client';

import {
  addDays,
  differenceInMinutes,
  format,
  isSameDay,
  startOfWeek,
} from 'date-fns';
import { ChevronLeft, ChevronRight } from 'lucide-react';
import { useState } from 'react';

import { Button } from './Button';

interface Booking {
  id: string;
  start: Date;
  end: Date;
  color: 'blue' | 'red' | 'green' | 'yellow';
}

interface WeeklyCalendarProps {
  bookings?: Booking[];
  onNavigate?: (date: Date) => void;
  initialDate?: Date; // Add this prop
}

export function WeeklyCalendar({
  bookings = [],
  onNavigate,
  initialDate = new Date(), // Use initialDate with default
}: WeeklyCalendarProps) {
  const [currentDate, setCurrentDate] = useState(initialDate);
  const startOfCurrentWeek = startOfWeek(currentDate, { weekStartsOn: 1 });

  const weekDates = Array.from({ length: 7 }, (_, i) =>
    addDays(startOfCurrentWeek, i),
  );
  const hours = Array.from({ length: 24 }, (_, i) => i);

  const navigateWeek = (direction: 'prev' | 'next') => {
    const newDate = addDays(currentDate, direction === 'prev' ? -7 : 7);
    setCurrentDate(newDate);
    onNavigate?.(newDate);
  };

  const getBookingsForDate = (date: Date) => {
    return bookings.filter((booking) => isSameDay(date, booking.start));
  };

  return (
    <div className="relative mx-auto h-full w-full max-w-7xl overflow-hidden bg-gray-900 text-gray-100 shadow-2xl">
      <div className="flex items-center justify-between border-b border-gray-800 p-6">
        <h2 className="bg-gradient-to-r from-blue-400 to-purple-500 bg-clip-text text-2xl font-bold text-transparent">
          {format(weekDates[0], 'd')} â€” {format(weekDates[6], 'd MMM yyyy')}
        </h2>
        <div className="flex gap-2">
          <Button
            variant="outline"
            size="sm"
            onClick={() => setCurrentDate(new Date())}
            className="border-gray-700 bg-gray-800 text-gray-100 hover:bg-gray-700"
          >
            Today
          </Button>
          <Button
            variant="outline"
            size="icon"
            onClick={() => navigateWeek('prev')}
            className="border-gray-700 bg-gray-800 text-gray-100 hover:bg-gray-700"
          >
            <ChevronLeft className="h-4 w-4" />
          </Button>
          <Button
            variant="outline"
            size="icon"
            onClick={() => navigateWeek('next')}
            className="border-gray-700 bg-gray-800 text-gray-100 hover:bg-gray-700"
          >
            <ChevronRight className="h-4 w-4" />
          </Button>
        </div>
      </div>

      <div className="overflow-x-auto">
        <div className="grid min-w-[800px] grid-cols-[auto_repeat(7,_minmax(120px,_1fr))]">
          {/* Time column */}
          <div className="sticky left-0 z-10 bg-gray-900">
            <div className="h-14 border-b border-gray-800" />
            {hours.map((hour) => (
              <div
                key={hour}
                className="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
              >
                {format(new Date().setHours(hour, 0), 'HH:mm')}
              </div>
            ))}
          </div>

          {/* Days columns */}
          {weekDates.map((date) => (
            <div key={date.toString()}>
              <div className="flex h-14 items-center justify-center border-b border-gray-800 px-2 font-medium">
                <div className="text-center">
                  <div className="text-gray-400">{format(date, 'EEE')}</div>
                  <div className="text-lg">{format(date, 'd')}</div>
                </div>
              </div>
              <div className="relative">
                {hours.map((hour) => (
                  <div
                    key={hour}
                    className="group relative h-16 border-b border-r border-gray-800"
                  />
                ))}
                {getBookingsForDate(date).map((booking) => {
                  const startOfDay = new Date(booking.start).setHours(
                    0,
                    0,
                    0,
                    0,
                  );
                  const top =
                    (differenceInMinutes(booking.start, startOfDay) / 60) * 64; // 64px = height of hour cell
                  const height =
                    (differenceInMinutes(booking.end, booking.start) / 60) * 64;

                  return (
                    <div
                      key={booking.id}
                      style={{
                        top: `${top}px`,
                        height: `${height}px`,
                      }}
                      className={`absolute left-0 right-0 mx-1 flex items-center overflow-hidden rounded-md px-2 text-xs font-medium ${booking.color === 'blue' ? 'bg-blue-500 text-white' : ''} ${booking.color === 'red' ? 'bg-red-500 text-white' : ''} ${booking.color === 'green' ? 'bg-green-500 text-white' : ''} ${booking.color === 'yellow' ? 'bg-yellow-500 text-gray-900' : ''} `}
                    >
                      <div className="truncate">
                        {format(booking.start, 'HH:mm')} -{' '}
                        {format(booking.end, 'HH:mm')}
                      </div>
                    </div>
                  );
                })}
              </div>
            </div>
          ))}
        </div>
      </div>
    </div>
  );
}
