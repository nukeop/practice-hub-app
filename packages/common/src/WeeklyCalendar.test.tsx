import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { addHours } from 'date-fns';

import { WeeklyCalendar } from './WeeklyCalendar';

const mockDate = new Date('2024-01-15T00:00:00.000Z'); // A Monday

describe('WeeklyCalendar', () => {
  it('should always be UTC', () => {
    expect(new Date().getTimezoneOffset()).toBe(0);
  });

  it('renders correctly with default props', () => {
    const { container } = render(<WeeklyCalendar initialDate={mockDate} />);
    expect(container).toMatchSnapshot();
  });

  it('displays correct date range in header', () => {
    render(<WeeklyCalendar initialDate={mockDate} />);
    expect(screen.getByText(/15 — 21 Jan 2024/)).toBeInTheDocument();
  });

  it('displays bookings correctly', () => {
    const bookings = [
      {
        id: '1',
        start: addHours(mockDate, 2), // 2 AM on Monday
        end: addHours(mockDate, 4), // 4 AM on Monday
        color: 'blue' as const,
      },
    ];

    render(<WeeklyCalendar initialDate={mockDate} bookings={bookings} />);

    expect(screen.getByText('02:00 - 04:00')).toBeInTheDocument();
  });

  it('returns to today when clicking Today button', () => {
    render(<WeeklyCalendar initialDate={mockDate} />);

    const todayButton = screen.getByRole('button', { name: /today/i });
    userEvent.click(todayButton);

    expect(screen.getByText(/15 — 21 Jan 2024/)).toBeInTheDocument();
  });

  it('navigates to next week when clicking Next button', async () => {
    const onNavigate = jest.fn();
    render(<WeeklyCalendar initialDate={mockDate} onNavigate={onNavigate} />);

    const nextButton = screen.getByTestId('next-week');
    await userEvent.click(nextButton);

    expect(onNavigate).toHaveBeenCalledWith(addHours(mockDate, 7 * 24));
  });

  it('navigates to previous week when clicking Previous button', async () => {
    const onNavigate = jest.fn();
    render(<WeeklyCalendar initialDate={mockDate} onNavigate={onNavigate} />);

    const prevButton = screen.getByTestId('prev-week');
    await userEvent.click(prevButton);

    expect(onNavigate).toHaveBeenCalledWith(addHours(mockDate, -7 * 24));
  });

  it('displays all 24 hours', () => {
    render(<WeeklyCalendar initialDate={mockDate} />);
    expect(screen.getByText('00:00')).toBeInTheDocument();
    expect(screen.getByText('23:00')).toBeInTheDocument();
  });

  it('displays all weekdays', () => {
    render(<WeeklyCalendar initialDate={mockDate} />);
    ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'].forEach((day) => {
      expect(screen.getByText(day)).toBeInTheDocument();
    });
  });
});
