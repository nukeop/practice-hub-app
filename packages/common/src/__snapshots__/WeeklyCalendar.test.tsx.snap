// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeeklyCalendar renders correctly with default props 1`] = `
<div>
  <div
    class="relative mx-auto h-full w-full max-w-7xl overflow-hidden bg-gray-900 text-gray-100 shadow-2xl"
  >
    <div
      class="flex items-center justify-between border-b border-gray-800 p-6"
    >
      <h2
        class="bg-gradient-to-r from-blue-400 to-purple-500 bg-clip-text text-2xl font-bold text-transparent"
      >
        15
         â€” 
        21 Jan 2024
      </h2>
      <div
        class="flex gap-2"
      >
        <button
          class="rounded-full px-6 py-3 font-bold transition-colors border-stone-800 text-stone-400 hover:bg-stone-900 hover:text-stone-200 border-2 bg-transparent border-gray-700 bg-gray-800 text-gray-100 hover:bg-gray-700"
        >
          Today
        </button>
        <button
          class="rounded-full px-6 py-3 font-bold transition-colors border-stone-800 text-stone-400 hover:bg-stone-900 hover:text-stone-200 border-2 bg-transparent border-gray-700 bg-gray-800 text-gray-100 hover:bg-gray-700"
          data-testid="prev-week"
        >
          <svg
            class="lucide lucide-chevron-left h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m15 18-6-6 6-6"
            />
          </svg>
        </button>
        <button
          class="rounded-full px-6 py-3 font-bold transition-colors border-stone-800 text-stone-400 hover:bg-stone-900 hover:text-stone-200 border-2 bg-transparent border-gray-700 bg-gray-800 text-gray-100 hover:bg-gray-700"
          data-testid="next-week"
        >
          <svg
            class="lucide lucide-chevron-right h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9 18 6-6-6-6"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="overflow-x-auto"
    >
      <div
        class="grid min-w-[800px] grid-cols-[auto_repeat(7,_minmax(120px,_1fr))]"
      >
        <div
          class="sticky left-0 z-10 bg-gray-900"
        >
          <div
            class="h-14 border-b border-gray-800"
          />
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            00:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            01:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            02:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            03:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            04:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            05:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            06:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            07:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            08:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            09:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            10:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            11:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            12:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            13:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            14:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            15:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            16:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            17:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            18:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            19:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            20:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            21:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            22:00
          </div>
          <div
            class="flex h-16 items-center border-b border-r border-gray-800 px-2 text-gray-400"
          >
            23:00
          </div>
        </div>
        <div>
          <div
            class="flex h-14 items-center justify-center border-b border-gray-800 px-2 font-medium"
          >
            <div
              class="text-center"
            >
              <div
                class="text-gray-400"
              >
                Mon
              </div>
              <div
                class="text-lg"
              >
                15
              </div>
            </div>
          </div>
          <div
            class="relative"
          >
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
          </div>
        </div>
        <div>
          <div
            class="flex h-14 items-center justify-center border-b border-gray-800 px-2 font-medium"
          >
            <div
              class="text-center"
            >
              <div
                class="text-gray-400"
              >
                Tue
              </div>
              <div
                class="text-lg"
              >
                16
              </div>
            </div>
          </div>
          <div
            class="relative"
          >
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
          </div>
        </div>
        <div>
          <div
            class="flex h-14 items-center justify-center border-b border-gray-800 px-2 font-medium"
          >
            <div
              class="text-center"
            >
              <div
                class="text-gray-400"
              >
                Wed
              </div>
              <div
                class="text-lg"
              >
                17
              </div>
            </div>
          </div>
          <div
            class="relative"
          >
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
          </div>
        </div>
        <div>
          <div
            class="flex h-14 items-center justify-center border-b border-gray-800 px-2 font-medium"
          >
            <div
              class="text-center"
            >
              <div
                class="text-gray-400"
              >
                Thu
              </div>
              <div
                class="text-lg"
              >
                18
              </div>
            </div>
          </div>
          <div
            class="relative"
          >
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
          </div>
        </div>
        <div>
          <div
            class="flex h-14 items-center justify-center border-b border-gray-800 px-2 font-medium"
          >
            <div
              class="text-center"
            >
              <div
                class="text-gray-400"
              >
                Fri
              </div>
              <div
                class="text-lg"
              >
                19
              </div>
            </div>
          </div>
          <div
            class="relative"
          >
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
          </div>
        </div>
        <div>
          <div
            class="flex h-14 items-center justify-center border-b border-gray-800 px-2 font-medium"
          >
            <div
              class="text-center"
            >
              <div
                class="text-gray-400"
              >
                Sat
              </div>
              <div
                class="text-lg"
              >
                20
              </div>
            </div>
          </div>
          <div
            class="relative"
          >
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
          </div>
        </div>
        <div>
          <div
            class="flex h-14 items-center justify-center border-b border-gray-800 px-2 font-medium"
          >
            <div
              class="text-center"
            >
              <div
                class="text-gray-400"
              >
                Sun
              </div>
              <div
                class="text-lg"
              >
                21
              </div>
            </div>
          </div>
          <div
            class="relative"
          >
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
            <div
              class="group relative h-16 border-b border-r border-gray-800"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
